WordPress/gutenberg
# Components: Add eslint rule for theme var regressions

Part of #44116

## What?

Adds a Eslint rule for JS/TS files under `packages/components` to prevent unknowing usage of `--wp-admin-theme-*` color variables.

## Why?

`--wp-admin-theme-*` variables are coupled with wp-admin and do not support our component theming system.

This PR is the Eslint counterpart of the Stylelint rules proposed in #58098.

## Testing Instructions

1. Check out the first commit 64c659d4423636f2b72fcab95411e005bf5453cb that adds the eslint rule, and run `npm run lint:js`. It should error on the offending files under `packages/components`.
2. The lint should pass after all the files are fixed.
